src/hera/workflows/script.py

```diff
@@ -282,17 +318,72 @@ def _get_parameters_from_callable(source: Callable) -> List[Parameter]:
     return parameters
 
 
-def _get_parameters_and_artifacts_from_callable(
+def _get_outputs_from_callable(
     source: Callable,
 ) -> Tuple[List[Parameter], List[Artifact]]:
-    # If there are any kwargs arguments associated with the function signature,
-    # we store these as we can set them as default values for argo arguments
+    # Look through the function signature return and find all the parameters
+    # and artifacts defined there
+    parameters = []
+    artifacts = []
+
+    if get_origin(inspect.signature(source).return_annotation) is Annotated:
+        annotation = get_args(inspect.signature(source).return_annotation)[1]
+        if isinstance(annotation, Artifact):
+            artifacts.append(annotation)
+        elif isinstance(annotation, Parameter):
+            parameters.append(annotation)
+    elif get_origin(inspect.signature(source).return_annotation) is tuple:
+        for a in get_args(inspect.signature(source).return_annotation):
+            annotation = get_args(a)[1]
+            if isinstance(annotation, Artifact):
+                artifacts.append(annotation)
+            elif isinstance(annotation, Parameter):
+                parameters.append(annotation)
+
+    return parameters, artifacts
+
+
+def _get_outputs_from_callable_signature(source: Callable) -> Tuple[List[Parameter], List[Artifact]]:
+    # Look through the function signature and find all parameters and artifacts annotated as output
+    parameters = []
+    artifacts = []
+
+    for name, p in inspect.signature(source).parameters.items():
+        if get_origin(p.annotation) is not Annotated:
+            continue
+        annotation = get_args(p.annotation)[1]
+        if not hasattr(annotation, "output") or not annotation.output:
+            continue
+        mytype = type(annotation)
```

Comment:
Tiny name suggestion and then adding an assert helps tell us what that type should be and that calling `get_input_attributes` on it will be legal (side note, did we make that an abstract method or protocol somewhere?)

```suggestion
        output_type = type(annotation)
        assert isintance(output_type, (Artifact, Parameter))
```